File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/BuildUrlProviders.java
package com.lasthopesoftware.bluewater.client.connection.builder;

import com.lasthopesoftware.bluewater.client.connection.url.IUrlProvider;
import com.lasthopesoftware.bluewater.client.library.repository.Library;
import com.namehillsoftware.handoff.promises.Promise;

public interface BuildUrlProviders {
	Promise<IUrlProvider> promiseBuiltUrlProvider(final Library library);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/BuildUrlProviders.kt
package com.lasthopesoftware.bluewater.client.connection.builder

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.connection.ServerConnection
import com.namehillsoftware.handoff.promises.Promise

interface BuildUrlProviders {
	fun promiseBuiltUrlProvider(libraryId: LibraryId): Promise<ServerConnection?>
}
